lib/broadway/topology.ex

```diff
@@ -387,9 +387,12 @@ defmodule Broadway.Topology do
       module: module,
       context: context,
       terminator: terminator,
-      shutdown: shutdown
+      shutdown: shutdown,
+      producer_config: producer_config
     } = config
 
+    {producer, _} = Keyword.get(producer_config, :module, {nil, nil})
```

Comment:
Should we preserve the options and emit those as well?

Reply:
Can the producer ever be nil? I don’t think it can.

also, should we pass the producer to the processor events?

Reply:
I followed the same pattern for `Broadway.Topology.ProcessorStage` so producer is now in all of the `[:broadway, (:processor|:batch_processor) (:start|:stop)]` events